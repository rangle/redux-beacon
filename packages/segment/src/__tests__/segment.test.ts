import Segment from '../';

window.analytics = {
  identify: jest.fn(),
  group: jest.fn(),
  page: jest.fn(),
  track: jest.fn(),
  alias: jest.fn(),
};
const target = Segment();

describe('Target: Segment.io', () => {
  it('does not call any service when hitType is undefined', () => {
    const events = [
      {
        hitType: undefined,
      },
    ];
    target(events);
    Object.keys(window.analytics).forEach(key => {
      expect(window.analytics[key]).not.toHaveBeenCalled();
    });
  });

  it('calls track service when hitType is event', () => {
    const events = [
      {
        hitType: 'event',
        eventAction: 'random event action',
      },
    ];
    target(events);
    expect(window.analytics.track).toHaveBeenCalledWith(
      events[0].eventAction,
      events[0]
    );
  });
});

describe('When Segment is not defined', () => {
  it('should throw an error informing the user.', () => {
    window.analytics = undefined;
    expect(() => target()).toThrowErrorMatchingSnapshot();
  });
});
