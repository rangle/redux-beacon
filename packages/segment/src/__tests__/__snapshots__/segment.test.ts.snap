// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Segment is not defined should throw an error informing the user. 1`] = `"window.analytics is not defined, Have you forgotten to include the Segment tracking snippet?"`;
